superset/explore/form_data/commands/delete.py

```diff
@@ -46,14 +48,15 @@ def run(self) -> bool:
                 key
             )
             if state:
-                dataset_id = state["dataset_id"]
-                chart_id = state["chart_id"]
-                check_access(dataset_id, chart_id, actor)
+                datasource_id: int = state["datasource_id"]
+                chart_id: Optional[int] = state["chart_id"]
+                datasource_type: str = state["datasource_type"]
+                check_chart_access(datasource_id, chart_id, actor, datasource_type)
                 if state["owner"] != actor.get_user_id():
                     raise TemporaryCacheAccessDeniedError()
                 tab_id = self._cmd_params.tab_id
                 contextual_key = cache_key(
-                    session.get("_id"), tab_id, dataset_id, chart_id
+                    session.get("_id"), tab_id, datasource_id, chart_id
```

Comment:
wouldn't we need to add `datasource_type` or we would get the same cache hit for datasource_id but different datasource_type?

```suggestion
                    session.get("_id"), tab_id, datasource_id, datasource_id, chart_id
```

Reply:
yes, thanks for catching!